src/build/build_step.go

```diff
@@ -40,6 +41,42 @@ var httpClientOnce sync.Once
 
 var magicSourcesWorkerKey = "WORKER"
 
+var targetsBuilt = metrics.NewCounter(
+	"build",
+	"targets_built_total",
+	"Total number of non-filegroup targets built",
+)
+
+var filegroupsBuilt = metrics.NewCounter(
+	"build",
+	"filegroups_built_total",
+	"Total number of filegroups built",
+)
+
+var targetsUnbuilt = metrics.NewCounter(
+	"build",
+	"targets_unbuilt_total",
+	"Total number of targets that we didn't need to rebuild",
+)
```

Comment:
How about `targetsUnchanged` or `targetsReused` to borrow from the target/logging lingo? 

Also these are only the targets that were already in `plz-out`. Perhaps we also want a metric for those that were retrieved from the cache (L301). Seems like these are included in the built targets at the moment which doesn't seem that useful. Is the idea that you can infer this information by subtracting the cache hits?

Reply:
There are cache metrics for how many got retrieved.

Maybe it's better to have targetsBuilt just be the total and then we can subtract those others from it?

Reply:
> There are cache metrics for how many got retrieved.
> 
> Maybe it's better to have targetsBuilt just be the total and then we can subtract those others from it?

This is incremented only when we retrieved from `plz-out` though? We retrieve from the cache closer to line 300, at which point we've already hit `defer targetsBuilt.Inc()`:

```golang

		if state.Cache != nil && !runRemotely && !state.ShouldRebuild(target) {
			// Note that ordering here is quite sensitive since the post-build function can modify
			// what we would retrieve from the cache.
			if target.BuildCouldModifyTarget() {
				log.Debug("Checking for build metadata for %s in cache...", target.Label)
				if metadata = retrieveFromCache(state.Cache, target, cacheKey, nil); metadata != nil {
					addOutDirOutsFromMetadata(target, metadata)
					if target.PostBuildFunction != nil && !haveRunPostBuildFunction {
						postBuildOutput = string(metadata.Stdout)
						if err := runPostBuildFunction(tid, state, target, postBuildOutput, ""); err != nil {
							return err
						}
					}
					// Now that we've updated the rule, retrieve the artifacts with the new output hash
					if retrieveArtifacts(tid, state, target, oldOutputHash) {
						return writeRuleHash(state, target)
					}
				}
			} else if retrieveArtifacts(tid, state, target, oldOutputHash) {
				return nil
			}
		}
```